crates/cubecl-wgpu/src/compiler/spirv.rs

```diff
@@ -178,9 +179,12 @@ impl WgpuCompiler for SpirvCompiler<GLCompute> {
 fn request_device(
     wgpu_adapter: &wgpu::Adapter,
     adapter: &vulkan::Adapter,
-    features: Features,
+    mut features: Features,
     limits: Limits,
 ) -> (wgpu::Device, wgpu::Queue) {
+    // This registers only f16 but not u8/i8, so remove so we can manually add them
+    features.remove(Features::SHADER_F16);
```

Comment:
I think we should check if it is present before adding it to the device properties no? It is hardcoded as available in the `register_types` function, but it might not always be available?

Reply:
All the datatype features in Vulkan are mandatory as far as I can tell. I don't believe they're device specific, they're handled by the driver.